src/app/collective-rewards/rewards/builders/hooks/useGetBuildersRewards.ts

```diff
@@ -30,28 +32,39 @@ export type BuildersRewards = {
 }
 
 export const useGetBuildersRewards = ({ rif, rbtc }: { [token: string]: Token }, currency = 'USD') => {
-  const { data: builders, isLoading: searchLoading, error: searchError } = useGetBuildersByState()
-  const buildersV2 = builders as Required<Builder>[]
+  const {
+    initialState: { allocations },
+    state: { getBuilderIndexByAddress },
+  } = useContext(AllocationsContext)
+  let {
+    data: builders,
+    isLoading: searchLoading,
+    error: searchError,
+  } = useGetBuildersByState<Required<Builder>>()
+
+  const filteredBuilders = builders.filter(({ address, stateFlags }) => {
+    if (!stateFlags.kycApproved || stateFlags.revoked) {
+      const allocation = allocations[getBuilderIndexByAddress(address)]
+      if (!allocation || allocation === 0n) return false
+    }
+
+    return true
+  })
+
   const {
     data: totalPotentialRewards,
     isLoading: totalPotentialRewardsLoading,
     error: totalPotentialRewardsError,
   } = useGetTotalPotentialReward()
 
-  const buildersAddress = buildersV2.map(({ address }) => address)
+  const buildersAddress = builders.map(({ address }) => address)
```

Comment:
Should we use `filteredBuilders` here?

Reply:
@franciscotobar I've checked it and here I think we should use the filtered builders only since the pct is then used for them only